### PR TITLE
Windows PerfMon GPU Engine Usage

### PR DESCRIPTION
**Pull request by adrian-kong**
Adding Windows GPU Utilization Percentage as per #2112

---

Tried to draft up something basic logic/script verifying for GPU Engine / Utilization Percentage in PerfMon like the other PerfCounters but ran into some weird numbers (not sure if I made a mistake in the code...) 

---

Taking some instance `pid_10472_luid_0x00000000_0x0000f814_phys_0_eng_0_engtype_3d` that I found in `Get-Counter -Counter "\GPU Engine(*)\Utilization Percentage"`...

```shell
Get-Counter -Counter "\GPU Engine(pid_10472_luid_0x00000000_0x0000f814_phys_0_eng_0_engtype_3d)\Utilization Percentage"
```
gives `3.05119782746321` in powershell but using this script returns `2237473511`, changes a bit but looks like the result of:
```shell
Get-Counter -Counter "\GPU Engine(pid_10472_luid_0x00000000_0x0000f814_phys_0_eng_0_engtype_3d)\Running Time"
```

It also looks like using the WildcardQuery gives
```
{RUNNING_TIME=[2074738846, 0, 0, 0, 0, 11338102288, 0, 0, 0, 0, 0, 0, 0, 3196630, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5466, 0, 7160799, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 162739, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7607, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 39585, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 9453, 5273, 9142, 1321, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1886491788, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 52307, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 589916896, 0, 0, 0, 0, 0, 4294745, 0, 0, 0, 0, 0, 4881145, 0, 0, 0, 9161068, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16413, 0, 4948, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3607614, 0, 0, 0, 0, 7934, 0, 0, 0, 0, 0, 402, 764, 61286, 2460900533, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 115478, 0, 0, 75059346, 0, 0, 14940, 0, 0, 0, 0, 0, 22012, 0, 0, 0, 18956634, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1731002, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 13422, 0, 19054, 0, 0, 0, 0, 0, 0, 13177, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 201858519, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 771442, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 661929671, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3502, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 421490027, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 119978079, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 207149, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8067, 0, 879036, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5585, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 
UTILIZATION=[2074738846, 0, 0, 0, 0, 11338102288, 0, 0, 0, 0, 0, 0, 0, 3196630, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5466, 0, 7160799, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 162739, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7607, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 39585, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 9453, 5273, 9142, 1321, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1886491788, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 52307, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 589916896, 0, 0, 0, 0, 0, 4294745, 0, 0, 0, 0, 0, 4881145, 0, 0, 0, 9161068, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16413, 0, 4948, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3607614, 0, 0, 0, 0, 7934, 0, 0, 0, 0, 0, 402, 764, 61286, 2460900533, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 115478, 0, 0, 75059346, 0, 0, 14940, 0, 0, 0, 0, 0, 22012, 0, 0, 0, 18956634, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1731002, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 13422, 0, 19054, 0, 0, 0, 0, 0, 0, 13177, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 201858519, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 771442, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 661929671, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3502, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 421490027, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 119978079, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 207149, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8067, 0, 879036, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5585, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}
```
basically two identical lists.

Wasn't sure if I was doing something wrong here with the enums...

### CODE REVIEW

File: oshi-core/src/test/java/oshi/PerfMonGpuEngineTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        @Override
+        public String getCounter() {
+            return counter;
+        }
+    }
+
+    // hmmmmm individual instance dont look right......
+    public enum GpuEngineProperty2 implements PerfCounterQuery.PdhCounterProperty {
```
Comment (dbwiddis) [inline oshi-core/src/test/java/oshi/PerfMonGpuEngineTest.java:30]: You're mixing types.  Since you're doing a wildcard enum you don't want the PdhCounterProperty, you want PdhCounterWildcardProperty.

**Comment by dbwiddis** [comment]
> hmmm I think it's lookin better now with the ratios,
> I got two GPUs tho so would need to figure out which is which as well as the WMI

I see 8 rows so not sure which two GPUs that is, and it's also mostly 0 which is interesting.  I wonder if this is something we need to start a daemon thread to measure (see the Windows load average thread for an example).

WMI is also a key "backup" option.  For all the other counters we have WMI tables so there ought to be something like `Win32_PerfRawData_Perf<something>_Gpu`.  These tables are just WMI access to the same counter data, but tend to be more reliable in non-English environments.

**Comment by dbwiddis** [comment]
And the columns are instances? So there are two GPU instances and a whole lot of something else?

**Comment by dbwiddis** [comment]
Thanks for your work on this PR!  Sorry for the delay, I'll make a best effort to get to it by the end of the week.
